src/main/scala/com/abdulradi/opentracing/xray/v1/Format.scala

```diff
@@ -90,12 +135,19 @@ object Format {
       "version" -> obj.version.asJson
     )
   )
-  implicit val subSegmentFieldsEncoder: ObjectEncoder[SubsegmentFields] = dummy // TODO
+  implicit val subSegmentFieldsEncoder: ObjectEncoder[SubsegmentFields] = ObjectEncoder.instance[SubsegmentFields](obj =>
+    ("namespace" -> obj.namespace.asJson) +:
+      ("precursor_ids" -> obj.precursorIds.asJson) +:
+      ("http" -> obj.http.asJson) +:
+      ("aws" -> obj.aws.asJson) +: obj.sql.asJsonObject
```

Comment:
This will merge the `sql` object with it's parent object, this is wrong, we need to retain the `sql` key.
Please write a test case that exposes this problem before fixing it.

Reply:
Okay